app/controllers/concerns/search_result_helper.rb

```diff
@@ -0,0 +1,50 @@
+module SearchResultHelper
+  class << self
+    # A method that uses the CardComponentSerializer to create search results
+    #
+    # @param [Array<Object>] results an array of results that is used to create search results
+    def create_search_results(results)
+      results.entries.map do |entry|
+        data = {
+          'heading-text':   entry.title,
+          url:              entry.links.first,
+          'list-hint':      list_hint_data(entry),
+          'paragraph-text': entry.content
+        }
+
+        ComponentSerializer::CardComponentSerializer.new('card__search-result', data).to_h
+      end
+    end
+
+    private
+
+    def list_hint_data(entry)
+      {}.tap do |hash|
+        handle_hint_type_and_text(hash, entry)
+        hash[:url] = entry.formatted_url
+      end
+    end
+
+    def list_hint_type(entry)
```

Comment:
What do you think of something like this?

Also, drop the use of `.first` - it will bite you later, what if 'Beta' isn't returned first?

```ruby
def list_hint_type(entry)
  types = ['hint']
  types << 'theme--grey-4' unless entry.hint_types.includes?('Beta')

  types.join(' ')
end
```